src/complex/matrix.rs

```diff
@@ -0,0 +1,2290 @@
+use std::{
+    cmp::{max, min},
+    fmt,
+    ops::{Add, Div, Index, IndexMut, Mul, Neg, Sub},
+};
+
+use anyhow::{bail, Result};
+use matrixmultiply::CGemmOption;
+use num_complex::Complex;
+use rand_distr::num_traits::{One, Zero};
+
+use crate::{
+    fuga::{
+        nearly_eq, tab, ConcatenateError, InnerProduct, LinearOp, MatrixProduct, Norm, Normed,
+        Shape, Vector,
+    },
+    traits::{fp::FPComplexMatrix, mutable::ComplexMutMatrix},
+};
+
+/// R-like complex matrix structure
+///
+/// # Examples
+///
+/// ```rust
+/// use peroxide::fuga::*;
+/// use num_complex::Complex64;
+/// use peroxide::complex::matrix::ComplexMatrix;
+///
+/// let v1 = ComplexMatrix {
+/// data: vec![
+///     Complex64::new(1f64, 1f64),
+///     Complex64::new(2f64, 2f64),
+///     Complex64::new(3f64, 3f64),
+///     Complex64::new(4f64, 4f64),
+/// ],
+/// row: 2,
+/// col: 2,
+/// shape: Row,
+/// }; // [[1+1i,2+2i],[3+3i,4+4i]]
+/// ```
+#[derive(Debug, Clone, Default)]
+pub struct ComplexMatrix {
+    pub data: Vec<Complex<f64>>,
+    pub row: usize,
+    pub col: usize,
+    pub shape: Shape,
+}
+
+// =============================================================================
+// Various complex matrix constructor
+// =============================================================================
+
+/// R-like complex matrix constructor
+///
+/// # Examples
+/// ```rust
+/// #[macro_use]
+/// extern crate peroxide;
+/// use peroxide::fuga::*;
+/// use num_complex::Complex64;
```

Comment:
Suggestion: I already write an alias for Complex64 as C64 in complex/mod.rs.
In example, how about use C64?

Reply:
This is fine. But writing Complex64 makes it more obvious for the user. Either way it does not hurt.